java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -0,0 +1,253 @@
+import static io.fabric8.java.generator.nodes.Keywords.JAVA_UTIL_MAP;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.EnumDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+import io.fabric8.java.generator.exceptions.JavaGeneratorException;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JObject extends AbstractJSONSchema2Pojo {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JObject.class);
+    private static final Set<String> IGNORED_FIELDS = new HashSet<>();
+
+    static {
+        IGNORED_FIELDS.add("description");
+        IGNORED_FIELDS.add("schema");
+        IGNORED_FIELDS.add("example");
+        IGNORED_FIELDS.add("examples");
+    }
+
+    private final String type;
+    private final Map<String, AbstractJSONSchema2Pojo> fields;
+    private final Set<String> required;
+    private JObjectOptions options;
+
+    public JObject(
+            String type,
+            Map<String, JSONSchemaProps> fields,
+            List<String> required,
+            JObjectOptions options) {
+        this.options = options;
+        this.required = new HashSet<>();
+        this.fields = new HashMap<>();
+
+        if (required != null) {
+            this.required.addAll(required);
+        }
+
+        String nextPrefix = options.getPrefix();
+        String nextSuffix = options.getSuffix();
+
+        if (type.toLowerCase(Locale.ROOT).equals("spec")) {
+            nextPrefix = "";
+            nextSuffix = "Spec";
+        }
+
+        this.type =
+                AbstractJSONSchema2Pojo.sanitizeString(
+                        options.getPrefix()
+                                + type.substring(0, 1).toUpperCase()
+                                + type.substring(1)
+                                + options.getSuffix());
+
+        if (fields == null) {
+            // no fields
+        } else {
+            for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
+                if (!IGNORED_FIELDS.contains(field.getKey()))
+                    this.fields.put(
+                            field.getKey(),
+                            AbstractJSONSchema2Pojo.fromJsonSchema(
+                                    field.getKey(), field.getValue(), nextPrefix, nextSuffix));
+            }
+        }
+    }
+
+    @Override
+    public String getType() {
+        return this.type;
+    }
+
+    @Override
+    public GeneratorResult generateJava(CompilationUnit cu) {
```

Comment:
Feel like this method could be broken up

Reply:
yes, especially offloading all the annotations String would be very beneficial.
But this PR has been reviewed already by several ppl, we should broke this up in a follow up PR IMHO.